packages/components/src/mobile/audio-player/test/audio-url-parser.native.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Internal dependencies
+ */
+import { parseAudioUrl } from '../audio-url-parser.native';
+
+const supportedAudioUrlsWithExtensions = [
+	'https://www.mp3.com/file.mp3?key1=value1&key2=value2#anchorforsomereason',
+	'https://www.mp3.com/file.mp3?key1=value1&key2=value2',
+	'https://www.mp3.com/file.mp3',
+];
+
+const supportedAudioUrlsWithoutExtensions = [
+	'https://www.mp3.com/file?key1=value1&key2=value2#anchorforsomereason',
+	'https://www.mp3.com/file?key1=value1&key2=value2',
+	'https://www.mp3.com/file',
+	'https://www.mp3.com/folder/file',
+];
+
+describe( 'supportedAudioUrlsWithExtensions', () => {
+	supportedAudioUrlsWithExtensions.forEach( ( url ) => {
+		it( `supports ${ url }`, () => {
+			const { title, extension } = parseAudioUrl( url );
+			expect( title ).not.toBe( '' );
+			expect( extension ).not.toBe( '' );
```

Comment:
I'm wondering if we could check here that the title and extension matches specific values, for example:

URL: `https://www.mp3.com/file.mp3`
```
expect( title ).toBe( 'file' );
expect( extension ).toBe( 'MP3' );
```

This way we could also verify that the extension is upper case.

Reply:
Indeed. Verification for title and extension added e4ea54b